File: 2542 - Iu-Di-Oh!.py
while True:
    Mlist = []
    Llist = []

    try:
        N = int(input())
        ML = input()
        MLlist = ML.split()
        M = int(MLlist[0])
        L = int(MLlist[1])

        for x in range(M):
            a = input()
            b = a.split()
            Mlist.append(b)
        for x in range(L):
            a = input()
            b = a.split()
            Llist.append(b)

        CMCL = input()
        CmCl = CMCL.split()
        Cm = int(CmCl[0])
        Cl = int(CmCl[1])
        chosen = int(input())

        mm = int(Mlist[Cm-1][chosen-1])
        ll = int(Llist[Cl-1][chosen-1])

        if(mm > ll):
            print("Marcos")
        elif(mm < ll):
            print("Leonardo")
        else:
            print("Empate")


    except EOFError:
        break

File: 2542 - Iu-Di-Oh!.c
#include <stdio.h>
int main()
{
    int i,j,N,M,L,M_C,L_C,chosen;
    int result;

    while(scanf("%d", &N) >= 0){
        scanf("%d %d",&M,&L);
        int M_d[M][N], L_d[L][N];

        for(i=0;i<M;i++){
            for(j=0;j<N;j++){
                scanf("%d",&M_d[i][j]);
            }
        }
        for(i=0;i<L;i++){
            for(j=0;j<N;j++){
                scanf("%d",&L_d[i][j]);
            }
        }

        scanf("%d %d %d",&M_C, &L_C, &chosen);

        if(M_d[M_C-1][chosen-1] > L_d[L_C-1][chosen-1])
            printf("Marcos\n");
        else if(M_d[M_C-1][chosen-1] < L_d[L_C-1][chosen-1])
            printf("Leonardo\n");
        else
            printf("Empate\n");


    }

    return 0;
}
